ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -470,12 +470,18 @@ private static void runTool(
                     Map<String, String> llmToolTmpParameters = new HashMap<>();
                     llmToolTmpParameters.putAll(tmpParameters);
                     llmToolTmpParameters.putAll(toolSpecMap.get(action).getParameters());
+                    if (toolSpecMap.get(action).getConfigMap() != null) {
+                        llmToolTmpParameters.putAll(toolSpecMap.get(action).getConfigMap());
+                    }
                     llmToolTmpParameters.put(MLAgentExecutor.QUESTION, actionInput);
                     tools.get(action).run(llmToolTmpParameters, toolListener); // run tool
                 } else {
                     Map<String, String> parameters = new HashMap<>();
                     parameters.putAll(tmpParameters);
                     parameters.putAll(toolParams);
+                    if (toolSpecMap.get(action).getConfigMap() != null) {
+                        parameters.putAll(toolSpecMap.get(action).getConfigMap());
+                    }
```

Comment:
I would suggest extract` parameters.putAll(toolSpecMap.get(action).getConfigMap());` outside of the if-else statement.

Reply:
Don't get it. I need to add the config parameters override immediately before tool run.